webapp/apifuncs.py

```diff
@@ -239,27 +199,37 @@ def add_todays_batting(result1, result2, playercode, fullname, pos, clubname):
     add stats from each game, player might have data in both, neither or just 1
     """
 
+    # establish stat translation table from MLB's stat keys to Rob's stat keys
+    bat_translation = {
+        'h': 'bhits',
+        'bb': 'bwalks',
+        'hr': 'hr',
+        'rbi': 'rbi',
+        'r': 'runs',
+        'ab': 'avg'
+    }
+
     # set initial values to have a basis for doing math
     game_stats = {}
-    for stat in ['bhits', 'bwalks', 'hr', 'rbi', 'runs', 'avg']:
-        game_stats[stat] = 0
+    for stat in bat_translation.keys():
+        game_stats[bat_translation[stat]] = 0
 
     # for each batting stat found update the associated variable
     # avg is field name in html but for todays stats will hold At Bats instead
     # column heading will reflect AVG or AB based on which stats are displayed
     if 'batting' in result1.keys():
-        game_stats = collate_stats(result1, game_stats)
+        game_stats = collate_stats(result1['batting'],
+                                   game_stats,
+                                   bat_translation)
     if 'batting' in result2.keys():
-        game_stats = collate_stats(result2, game_stats)
+        game_stats = collate_stats(result2['batting'],
+                                   game_stats,
+                                   bat_translation)
 
     # if all stats added together are at least 1 then player batted today
     # if all stats added together = 0 then he did not bat
-    if (game_stats['bhits'] +
-            game_stats['bwalks'] +
-            game_stats['hr'] +
-            game_stats['rbi'] +
-            game_stats['runs'] +
-            game_stats['avg']) > 0:
```

Comment:
These next 2 lines are really cool adding all entries in the game_stats dictionary in a single line without having to know the names of every entry in the stat pool. :)